starfish/imagestack/imagestack.py

```diff
@@ -889,6 +890,36 @@ def tile_metadata(self) -> pd.DataFrame:
 
         return pd.DataFrame(data)
 
+    @property
+    def log(self) -> List[dict]:
+        """
+        Returns a list of pipeline components that have been applied to this imagestack
+        as well as their corresponding runtime parameters.
+
+         ex.
+            [('GaussianHighPass', {'sigma': (3, 3), 'is_volume': False}),
+            ('GaussianLowPass', {'sigma': (1, 1), 'is_volume': False}])]
+
+        Means that this imagestack was created by applying a GaussianHighPass Filter then
+        a GaussianLowPass Filter to a starting imagestack
+
+        Returns
+        -------
+        List[Tuple[str, dict]]
+        """
+        return self._log
+
+    def update_log(self, class_instance):
```

Comment:
I think you can do `     def update_log(self, class_instance: AlgorithmBase): -> None`